pokemongo_bot/__init__.py

```diff
@@ -441,11 +444,16 @@ def tick(self):
 
         # Check if session token has expired
         self.check_session(self.position[0:2])
+        start_tick = time.time()
 
         for worker in self.workers:
             if worker.work() == WorkerResult.RUNNING:
                 return
 
+        end_tick = time.time()
```

Comment:
Right now we are already limiting our api wrapper to making at most two request to the api per second. Set here: https://github.com/PokemonGoF/PokemonGo-Bot/blob/master/pokemongo_bot/api_wrapper.py#L47

That is probably sufficient control for throttling to not need to have a minimum time for an entire tick.